### PR TITLE
Implement support for WebGPU viewFormats in surface configuration

### LINKED ISSUE #22939: WebGPU doesn't implement surface viewFormats
In src\library_webgpu.js there is this big block:

```js
#if ASSERTIONS
    var viewFormatCount = {{{ gpu.makeGetU32('config', C_STRUCTS.WGPUSurfaceConfiguration.viewFormatCount) }}};
    var viewFormats = {{{ makeGetValue('config', C_STRUCTS.WGPUSurfaceConfiguration.viewFormats, '*') }}};
    assert(viewFormatCount === 0 && viewFormats === 0, "TODO: Support viewFormats.");
    var alphaMode = {{{ gpu.makeGetU32('config', C_STRUCTS.WGPUSurfaceConfiguration.alphaMode) }}};
    assert(alphaMode === {{{ gpu.CompositeAlphaMode.Auto }}} ||
      alphaMode === {{{ gpu.CompositeAlphaMode.Opaque }}},
      "TODO: Support WGPUCompositeAlphaMode_Premultiplied.");
    assert({{{ gpu.PresentMode.Fifo }}} ===
      {{{ gpu.makeGetU32('config', C_STRUCTS.WGPUSurfaceConfiguration.presentMode) }}});
#endif
```

The `TODO: Support viewFormats` assertion is of particular interest to me. Without specifying compatible view formats for the backbuffer, it is not possible to use sRGB textures, which means I can't do [Gamma correct rendering](https://learnopengl.com/Advanced-Lighting/Gamma-Correction) in the browser.

This should only be a matter of passing the values from C++ to JS, 

In fact, this is already implemented during texture creation in `wgpuDeviceCreateTexture`:

```js
var viewFormatCount = {{{ gpu.makeGetU32('descriptor', C_STRUCTS.WGPUTextureDescriptor.viewFormatCount) }}};
if (viewFormatCount) {
  var viewFormatsPtr = {{{ makeGetValue('descriptor', C_STRUCTS.WGPUTextureDescriptor.viewFormats, '*') }}};
  // viewFormatsPtr pointer to an array of TextureFormat which is an enum of size uint32_t
  desc["viewFormats"] = Array.from({{{ makeHEAPView('32', 'viewFormatsPtr', `viewFormatsPtr + viewFormatCount * 4`) }}},
    function(format) { return WebGPU.TextureFormat[format]; });
}
```

I assume the fix would be very similar, but I'm no expert when it comes to JavaScript.

**Version of emscripten/emsdk:**

EMCC version:
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.65 (7f8a05dd4e37cbd7ffde6d624f91fd545f7b52e3)
clang version 20.0.0git (https:/github.com/llvm/llvm-project 547917aebd1e79a8929b53f0ddf3b5185ee4df74)
Target: wasm32-unknown-emscripten
Thread model: posix

### PR DESCRIPTION
**Pull request by jankrassnigg**
Fixes #22939.

Tested with a custom application on my machine.

### CODE REVIEW

File: src/library_webgpu.js [hunk 1]
```diff
@@ -2794,6 +2791,16 @@ var LibraryWebGPU = {
...
         {{{ gpu.makeGetU32('config', C_STRUCTS.WGPUSurfaceConfiguration.alphaMode) }}}],
     };
+
+    var viewFormatCount = {{{ gpu.makeGetU32('config', C_STRUCTS.WGPUSurfaceConfiguration.viewFormatCount) }}};
+
+    if (viewFormatCount) {
+      var viewFormats = {{{ makeGetValue('config', C_STRUCTS.WGPUSurfaceConfiguration.viewFormats, '*') }}};
+      // viewFormats pointer to an array of TextureFormat which is an enum of size uint32_t
+      configuration["viewFormats"] = Array.from({{{ makeHEAPView('32', 'viewFormats', `viewFormats + viewFormatCount * 4`) }}},
+        function(format) { return WebGPU.TextureFormat[format]; });
```
Comment (sbc100) [inline src/library_webgpu.js:2801]: Use a simple arrow function here?  `(format) => WebGPU.TextureFormat[format]`?

File: src/library_webgpu.js [hunk 2]
```diff
@@ -2794,6 +2791,16 @@ var LibraryWebGPU = {
       "alphaMode": WebGPU.AlphaMode[
         {{{ gpu.makeGetU32('config', C_STRUCTS.WGPUSurfaceConfiguration.alphaMode) }}}],
     };
+
+    var viewFormatCount = {{{ gpu.makeGetU32('config', C_STRUCTS.WGPUSurfaceConfiguration.viewFormatCount) }}};
+
+    if (viewFormatCount) {
+      var viewFormats = {{{ makeGetValue('config', C_STRUCTS.WGPUSurfaceConfiguration.viewFormats, '*') }}};
+      // viewFormats pointer to an array of TextureFormat which is an enum of size uint32_t
+      configuration["viewFormats"] = Array.from({{{ makeHEAPView('32', 'viewFormats', `viewFormats + viewFormatCount * 4`) }}},
```
Comment (sbc100) [inline src/library_webgpu.js:2800]: Use single quotes here

Comment (jankrassnigg) [reply src/library_webgpu.js:2800]: I just copied existing code from the same file. If you want the code to look different, feel free to modify the pr. To me, Java script is just gibberish that happens to sometimes do things. I'll leave all the rest up to you guys.

Comment (kainino0x) [reply src/library_webgpu.js:2800]: Sorry it's inconsistent! Happy to take care of these things.

File: src/library_webgpu.js (hunk 1 above)
Comment (kainino0x) [reply src/library_webgpu.js:2801]:
```suggestion
        format => WebGPU.TextureFormat[format]);
```

File: src/library_webgpu.js (hunk 2 above)
Comment (kainino0x) [reply src/library_webgpu.js:2800]: This still won't be consistent but at least it's closer to the final target :)
```suggestion
      configuration['viewFormats'] = Array.from({{{ makeHEAPView('32', 'viewFormats', `viewFormats + viewFormatCount * 4`) }}},
```

File: src/library_webgpu.js [hunk 3]
```diff
@@ -2794,6 +2791,16 @@ var LibraryWebGPU = {
       "alphaMode": WebGPU.AlphaMode[
         {{{ gpu.makeGetU32('config', C_STRUCTS.WGPUSurfaceConfiguration.alphaMode) }}}],
     };
+
+    var viewFormatCount = {{{ gpu.makeGetU32('config', C_STRUCTS.WGPUSurfaceConfiguration.viewFormatCount) }}};
+
+    if (viewFormatCount) {
+      var viewFormats = {{{ makeGetValue('config', C_STRUCTS.WGPUSurfaceConfiguration.viewFormats, '*') }}};
+      // viewFormats pointer to an array of TextureFormat which is an enum of size uint32_t
+      configuration['viewFormats'] = Array.from({{{ makeHEAPView('32', 'viewFormats', `viewFormats + viewFormatCount * 4`) }}},
```
Comment (kainino0x) [inline src/library_webgpu.js:2800]: Oops I missed one
```suggestion
      configuration['viewFormats'] = Array.from({{{ makeHEAPView('32', 'viewFormats', 'viewFormats + viewFormatCount * 4') }}},
```

**Review by kainino0x — APPROVED** [review]
lgtm, thank you!

**Pull request merged**